pkg/nvpci/nvpci.go

```diff
@@ -265,13 +303,25 @@ func (p *nvpci) GetGPUByPciBusID(address string) (*NvidiaPCIDevice, error) {
 		return nil, fmt.Errorf("unable to detect iommu_group for %s: %v", address, err)
 	}
 
+	var sriovInfo *SriovInfo
 	// device is a virtual function (VF) if "physfn" symlink exists.
-	var isVF bool
-	_, err = filepath.EvalSymlinks(path.Join(devicePath, "physfn"))
+	physFnAddress, err := filepath.EvalSymlinks(path.Join(devicePath, "physfn"))
 	if err == nil {
-		isVF = true
-	}
-	if err != nil && !os.IsNotExist(err) {
+		physFn, err := p.getGPUByPciBusID(filepath.Base(physFnAddress), cache)
+		if err != nil {
+			return nil, fmt.Errorf("unable to detect physfn for %s: %v", address, err)
+		}
+		sriovInfo = &SriovInfo{
+			VirtualFunction: &SriovVirtualFunction{
+				PhysicalFunction: physFn,
+			},
+		}
+	} else if os.IsNotExist(err) {
+		sriovInfo, err = p.getSriovInfoForPhysicalFunction(devicePath)
+		if err != nil {
+			return nil, err
+		}
+	} else {
 		return nil, fmt.Errorf("unable to resolve %s: %v", path.Join(devicePath, "physfn"), err)
 	}
```

Comment:
Let's move all of this to the bottom, right above where we put the device in the cache if it is a PF

Reply:
I don't know if we can as `IsPf` determines is device is physical function based on `SriovInfo`